pydataproc/dataproc.py

```diff
@@ -58,29 +71,34 @@ def cluster_bucket(self, cluster_name):
         :param cluster_name: The name of the cluster to check
         :return: staging bucket, or None if no such cluster
         """
-        clusters = self.list_clusters()
-        for c in clusters['clusters']:
-            if cluster_name == c['clusterName']:
-                return c['config']['configBucket']
+        for c, ci in self.list_clusters(minimal=False):
+            if c == cluster_name:
+                return ci['config']['configBucket']
 
         return None
 
-    def list_clusters(self):
+    def list_clusters(self, minimal=True):
         """
-        Queries the DataProc API, returning a list of all currently active clusters,
-        with all available details/configuration.
+        Queries the DataProc API, returning a dict of all currently active clusters,
+        keyed by cluster name.
+
+        If 'minimal' is specified, each cluster's current state will be returned,
+        otherwise the full cluster configuration will be returned.
 
+        :param minimal: reduces
         :return: list of dicts of cluster configuration
         """
         result = self.dataproc.projects().regions().clusters().list(
             projectId=self.project,
             region=self.region).execute()
-        return result
+        if minimal:
+            return {c['clusterName']: c['status']['state'] for c in result.get('clusters', [])}
+        return {c['clusterName']: c for c in result.get('clusters', [])}
 
     # TODO add support for preemptible workers
     def create_cluster(self, cluster_name, num_masters=1, num_workers=2,
                        master_type='n1-standard-1', worker_type='n1-standard-1',
-                       master_disk_gb=50, worker_disk_gb=50, init_script=None):
+                       master_disk_gb=50, worker_disk_gb=50, init_script=None, block=True):
```

Comment:
A log warning about the possibility of it being stuck if it errors out during initialization would be helpful.

Reply:
Ah, yes, nice one, ty